src/sklearn_evaluation/plot/feature_ranking.py

```diff
@@ -0,0 +1,410 @@
+            number of features. E.g. for 1D ranking, it is (n,), for a
+            2D ranking it is (n,n) and so forth.
+        """
+
+        algorithm = self.algorithm.lower()
+
+        if algorithm not in self.ranking_methods:
+            raise ValueError(
+                "'{}' is unrecognized ranking method".format(algorithm))
+
+        # Extract matrix from dataframe if necessary
+        if isinstance(X, pd.DataFrame):
+            X = X.values
+
+        return self.ranking_methods[algorithm](X)
+
+    def _derive_features_from_data(self, X):
+        n_columns = X.shape[1]
+
+        if self.features is not None:
+            if len(self.features) != n_columns:
+                raise ValueError((
+                    "number of supplied feature names does not match the number "
+                    "of columns in the training data."))
+
+            self.features_ = np.array(self.features)
+
+        else:
+            # Attempt to determine the feature names from the input data
+            if isinstance(X, pd.DataFrame):
+                self.features_ = np.array(X.columns)
+
+            # Otherwise create numeric labels for each column.
+            else:
+                self.features_ = np.arange(0, n_columns)
+
+    def _derive_features_from_ranks(self, ranks):
+        if self.features is None:
+            self.features_ = np.arange(0, len(ranks))
+        else:
+            if len(self.features) != len(ranks):
+                raise ValueError((
+                    "number of supplied feature names does not match the number "
+                    "of ranks provided."))
+            self.features_ = np.array(self.features)
+
+    @SKLearnEvaluationLogger.log(feature='plot')
+    def feature_ranks(self, X):
+        """
+        Parameters
+        ----------
+        X : array-like, shape (n_samples, n_features)
+            Feature dataset to be ranked.
+            Refer https://numpy.org/doc/stable/glossary.html#term-array-like
+
+        Returns
+        ------
+        ax: matplotlib Axes
+            Axes containing the plot
+        """
+        self._derive_features_from_data(X)
+        self.ranks_ = self._rank(X)
+        self._draw()
+        return self.ax
+
+    @SKLearnEvaluationLogger.log(feature='plot')
+    def feature_ranks_custom_algorithm(self, ranks):
+        """
+        This method is useful if user wants to use custom algorithm for feature ranking.
+
+        Parameters
+        ----------
+        ranks : ndarray
+            An n-dimensional, symmetric array of rank scores, where n is the
+            number of features. E.g. for 1D ranking, it is (n,), for a
+            2D ranking it is (n,n).
+
+        Returns
+        ------
+        ax: matplotlib Axes
+            Axes containing the plot
+        """
+
+        self._validate_rank(ranks)
+        self._derive_features_from_ranks(ranks)
+        self.ranks_ = ranks
+        self._draw()
+        return self.ax
+
+
+class Rank1D(RankD):
```

Comment:
The classes look very similar, does it make sense to use an abstract class/interface here?

Reply:
`RankD` is the base class and has all the common functionalities. The only different implementation is `_draw` method, I can move it as abstract to the base class.

Reply:
Would add `draw` to RankD for more possible implementations